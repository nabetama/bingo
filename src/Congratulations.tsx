import React from "react";

const Congratulations: React.FC<{
  isBingo: boolean;
}> = (props) => {
  if (props.isBingo) {
    return (
      <div
        className={
          "col-span-5 text-center font-serif text-3xl font-black text-red-600"
        }
      >
        BINGO!!
      </div>
    );
  } else {
    return <></>;
  }
};

export { Congratulations };
