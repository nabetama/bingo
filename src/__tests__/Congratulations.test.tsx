import React from "react";

import renderer from "react-test-renderer";
import { Congratulations } from "../Congratulations";

describe("<Congratulations />", () => {
  test("is not BINGO", () => {
    const component = renderer.create(<Congratulations isBingo={false} />);
    expect(component).toMatchSnapshot();
  });

  test("is BINGO", () => {
    const component = renderer.create(<Congratulations isBingo={true} />);
    expect(component).toMatchSnapshot();
  });
});
