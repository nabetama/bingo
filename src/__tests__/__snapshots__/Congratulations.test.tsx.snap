// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Congratulations /> is BINGO 1`] = `
<div
  className="col-start-3 col-end-3 text-center font-serif text-3xl font-black text-red-600"
>
  BINGO!!
</div>
`;

exports[`<Congratulations /> is not BINGO 1`] = `null`;
